openlibrary/plugins/upstream/borrow.py

```diff
@@ -155,19 +155,23 @@ def POST(self, key):
                     resource_type=resource_type,
                     user_key=user.key,
                     book_key=key)
-                loan_link = loan['loan_link']
 
-                if resource_type == 'bookreader':
-                    stats.increment('ol.loans.bookreader')
-                elif resource_type == 'pdf':
-                    stats.increment('ol.loans.pdf')
-                elif resource_type == 'epub':
-                    stats.increment('ol.loans.epub')
+                if loan:
+                    loan_link = loan['loan_link']
 
-                if resource_type == 'bookreader':
-                    raise web.seeother(make_bookreader_auth_link(loan.get_key(), edition.ocaid, '/stream/' + edition.ocaid, ol_host))
+                    if resource_type == 'bookreader':
+                        stats.increment('ol.loans.bookreader')
+                    elif resource_type == 'pdf':
+                        stats.increment('ol.loans.pdf')
+                    elif resource_type == 'epub':
+                        stats.increment('ol.loans.epub')
+
+                    if resource_type == 'bookreader':
```

Comment:
Why not do the stats.increment('ol.loans.bookreader') in this same if block rather than repeating the check in 162 and 169?